handle error events on xkeys that can be emitted during the init process before the instance is exposed for listener registration

loucadufault opened the pull request:
Fixes https://github.com/SuperFlyTV/xkeys/issues/112 (implements the first approach discussed)

loucadufault commented on packages/webhid/src/methods.ts:
```diff
@@ -69,13 +69,28 @@ export async function setupXkeysPanel(browserDevice: HIDDevice): Promise<XKeys>
+				.finally(() => (alreadyRejected = true))
+		}
+	    // Handle all error events until the instance is returned
+		xkeys.on('error', xkeysStopgapErrorHandler)
```
we don't use `once` intentionally. Until we have resolved back the instance to the caller, we are responsible to keep catching errors that may be emitted.

loucadufault commented:
Implementation could be revised with the addition of a `flush` method as discussed [here](https://github.com/SuperFlyTV/xkeys/issues/106#issuecomment-2310210337).

nytamin commented on packages/webhid/src/methods.ts:
```diff
@@ -69,13 +69,28 @@ export async function setupXkeysPanel(browserDevice: HIDDevice): Promise<XKeys>
+				return
+			}
+			deviceWrap.close()
+				.then(() => reject())
```
Minor note: is there a reason the `reject()` has no argument here? Should we `reject(e)`?

loucadufault replied:
This is for the path where `deviceWrap.close()` promise resolves, where it should resolve to a void promise, but still I want to avoid forwarding whatever value it resolves to as the rejected value of the enclosing (returned) promise in case the API of that method changes.

nytamin reviewed (commented):
Thanks for this PR! I agree, this should handle the edge case where an error event is emitted during initialization.

Note to self: When merging this, add a equivalent handling to `setupXKeysPanel` in the node-package.

loucadufault commented:
I added an implementation for node in #114.

I also simplified the implementation here and there to not handle the cleanup inside the promise executor, and instead rely on the existing catch clause which execution will break out to if the promise rejects.

loucadufault commented on packages/webhid/src/methods.ts:
```diff
@@ -69,13 +69,30 @@ export async function setupXkeysPanel(browserDevice: HIDDevice): Promise<XKeys>
+				.close()
+				.then(() => reject())
+				.catch(reject)
+				.finally(() => (alreadyRejected = true))
```
doing this in the `finally` (old implementation) was not semantically coherent since the flag only makes sense when the promise rejects, not when it resolves (same applies below). So changed this.
